cpp/src/arrow/visitor_inline.h

```diff
@@ -474,4 +474,45 @@ inline Status VisitScalarInline(const Scalar& scalar, VISITOR* visitor) {
 
 #undef TYPE_VISIT_INLINE
 
+// Visit a null bitmap, in order, without overhead.
+//
+// The given `VisitFunc` should be a callable with either of these signatures:
+// - void(bool is_valid)
+// - Status(bool is_valid)
```

Comment:
This doesn't seem worthwhile given that we have `internal::VisitBits` and `internal::VisitBitsUnrolled`. The Status return signature is not useful in `parquet::` since we can throw an exception to effect early termination

Reply:
Well, this function also gracefully separates the case where `null_count` is 0 (and the null bitmap pointer potentially null), both for correctness and better performance.